diff --git org.eclipse.mylyn.tasks/org.eclipse.mylyn.tasks.core/src/org/eclipse/mylyn/tasks/core/TaskInitializationData.java org.eclipse.mylyn.tasks/org.eclipse.mylyn.tasks.core/src/org/eclipse/mylyn/tasks/core/TaskInitializationData.java
index 39c3f1f..5bc9214 100644
--- org.eclipse.mylyn.tasks/org.eclipse.mylyn.tasks.core/src/org/eclipse/mylyn/tasks/core/TaskInitializationData.java
+++ org.eclipse.mylyn.tasks/org.eclipse.mylyn.tasks.core/src/org/eclipse/mylyn/tasks/core/TaskInitializationData.java
@@ -191,7 +191,6 @@ public class TaskInitializationData implements ITaskMapping {
 	/**
 	 * Throws {@link UnsupportedOperationException}.
 	 */
-	@Nullable
 	public void merge(ITaskMapping source) {
 		throw new UnsupportedOperationException();
 	}
@@ -248,4 +247,4 @@ public class TaskInitializationData implements ITaskMapping {
 		attributesById.setAttribute(TaskAttribute.VERSION, version);
 	}
 
-}
\ No newline at end of file
+}
