--- org.eclipse.mylyn.tasks/connector-bugzilla-rest/org.eclipse.mylyn.bugzilla.rest.core.tests/META-INF/MANIFEST.MF.orig       2015-01-29 14:13:38.896674915 +0000
+++ org.eclipse.mylyn.tasks/connector-bugzilla-rest/org.eclipse.mylyn.bugzilla.rest.core.tests/META-INF/MANIFEST.MF    2015-01-29 14:14:28.583954615 +0000
@@ -6,8 +6,8 @@ Bundle-Version: 1.0.0.qualifier
 Bundle-Vendor: Eclipse Mylyn
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Require-Bundle: org.junit;bundle-version="4.8.2",
- org.hamcrest;bundle-version="[1.0.0,1.2.0)",
- org.hamcrest.library;bundle-version="[1.0.0,1.2.0)",
+ org.hamcrest.core;bundle-version="[1.0.0,1.4.0)",
+ org.hamcrest.library;bundle-version="[1.0.0,1.4.0)",
  com.google.guava;bundle-version="15.0.0",
  com.google.gson;bundle-version="2.2.4",
  org.eclipse.core.runtime,
--- org.eclipse.mylyn.tasks/connector-bugzilla-rest/org.eclipse.mylyn.bugzilla.rest.tests/META-INF/MANIFEST.MF.orig	2015-01-29 15:00:50.434781602 +0000
+++ org.eclipse.mylyn.tasks/connector-bugzilla-rest/org.eclipse.mylyn.bugzilla.rest.tests/META-INF/MANIFEST.MF	2015-01-29 15:01:05.218570917 +0000
@@ -6,6 +6,7 @@
 Bundle-Vendor: Eclipse Mylyn
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Require-Bundle: org.junit;bundle-version="4.8.2",
+ org.hamcrest.core,
  org.eclipse.equinox.common,
  org.eclipse.jface,
  org.eclipse.ui.workbench,
